sig-and-wg/sig-core/proposals/knative_function_integration.md

```diff
@@ -0,0 +1,361 @@
+# Integration of the knative based function controller
+
+## Introduction
+
+One of the main features of kyma is its ability to run serverless functions. Currently the functions provider in Kyma is the open source project [kubeless](https://github.com/kubeless/kubeless). With the focus shift towards knative, Kubeless will be replaced with a functions runtime controller (called `function-controller` from here on) that is the provider of a [Knative](https://github.com/knative) based functions (called `knative-functions` from here on).
+
+The migration to a new functions provider should provide a similar user experience as with the currently available kubeless runtime. This includes UI based CRUD operations for the functions as well as monitoring, alerting and  backup / restore.
+
+## Features
+
+### Current Features
+
+- As a developer, I want to provide my code as a function of a certain language and runtime to be deployed.
+- As a developer, I want to specify the packages/library dependencies of my code so that it's available for my code.
+- As a developer, I want to be able to provide arbitrary environment variables as name, value pairs to be available for use inside my function.
+- As a developer, I want to create a `Service Binding` to a `Service Instance` that is provisioned inside my namespace.
+- As a developer, I want to expose my function via `http` & `https` with the option to add a token based authentication.
+- As a developer, I want to be able to choose different runtimes for my function.
+
+### New Features
+
+- As a developer, I want to define enable and disable scale to zero as well as specify the scale to zero grace period.
+- As a developer, I want to view the status of my function.
+
+## Proposed Solution
+
+### Overview
+We are proposing introducing the following changes:
+
+1. A new `Function` CRD and `function-controller` component.
+2. Bundle [Tekton Pipelines](https://github.com/tektoncd/pipeline/) and [Knative Serving](https://github.com/knative/serving) as Kyma modules.
+3. Bundle a docker registry inside Kyma for hosting function images.
+4. Adding GraphQL layer support.
+5. Updating the UI with a new Microfrontend Functions section that leverages all new functionality.
+
+### Function CRD
+
+#### CRD Definition
+
+The new `Function` CRD gives the user a way to define how to turn his source code into a running container inside Kyma. That source code is conventionally a single function running inside a specific execution environment (aka runtime). This is an example of the minimum set of fields you need to successfully define your function.
+
+```YAML
+apiVersion: functions.serverless.kyma-project.io/v1alpha1
```

Comment:
`serverless.kyma-project.io/v1alpha1` ?

Reply:
I copy pasted it from the code, but yeah I like that more.